ow_plexil/src/plexil-adapter/OwInterface.cpp

```diff
@@ -308,6 +303,14 @@ void OwInterface::antennaFaultCallback
   updateFaultStatus (msg->value, m_panTiltErrors, "ANTENNA", "AntennaFault");
 }
 
+static double normalize_degrees (double angle)
+{
+  static double pi = R2D * M_PI;
+  static double tau = pi * 2.0;
+
+  return fmod(angle + pi, tau) - pi;
```

Comment:
TIL C++ `fmod` != Python `%` (floating-point modulus)

When the numerator is negative, Python's floating-point modulus operator will use a negative quotient, so that the remainder that is returned is always positive. However, in the case of cmath's floating-point modulus function, the quotient is always selected to be positive, which means the remainder returned by the function will be negative so long as the denominator is positive. The result is that this function can return angles as low as -360. I think this is where the bug came from. This can be accounted for with the addition of an if-statement to handle the negative case. 
**Source:** https://stackoverflow.com/questions/11498169/dealing-with-angle-wrap-in-c-code
```suggestion
  double x = fmod(angle + pi, tau);
  if (x < 0)
    x += tau;
  return x - pi;
```

Reply:
Thanks, done.  This [StackOverflow article](https://stackoverflow.com/questions/11498169/dealing-with-angle-wrap-in-c-code) helped verify.